tasks/homework/homework.go

```diff
@@ -0,0 +1,184 @@
+package main
+
+import (
+	"fmt"
+)
+
+type Animal struct {
+	ID   int
+	Name string
+}
+
+type Areas map[string]Area
+
+type Area struct {
+	Name    string
+	Type    string
+	Sectors map[string]Sector
+}
+
+type Sector struct {
+	Subtype string
+	Animals []Animal
+}
+
+func (s *Sector) AddAnimal(animal Animal) {
+	s.Animals = append(s.Animals, animal)
+}
+
+func (s *Sector) MoveAnimal(animalID int, distSector *Sector) {
+	for i, animal := range s.Animals {
+		if animal.ID == animalID {
+			distSector.AddAnimal(animal)
+			s.Animals = append(s.Animals[:i], s.Animals[i+1:]...)
+			return
+		}
+	}
+}
+
+func (s *Sector) CleanUp() {
+	if s.Subtype == "technical" {
+		fmt.Println("technical sector cleaned up")
+	}
+}
+
+func (s *Sector) Feed(animal *Animal) {
+	if s.Subtype == "technical" {
+		fmt.Printf("feeding the %s\n", animal.Name)
+	}
+}
+
+type Zoo struct {
+	Areas Areas
+}
+
+func (z Zoo) Lookup(name string) *Sector {
+	return &Sector{}
+}
+
+func main() {
+	z := Zoo{
+		Areas: buildAreas(),
+	}
+
+	fmt.Println("Try to find by name Eagle")
+	eagle, err := FindAnimalByName(z.Areas, "Eagle")
+	if nil == err {
+		fmt.Printf("%s has ID: %d\n", eagle.Name, eagle.ID)
+	} else {
+		fmt.Println(err)
+	}
+
+	fmt.Println("Try to find by ID 8")
+	gorilla, err := FindAnimalByID(z.Areas, 8)
+	if nil == err {
+		fmt.Printf("Animal with ID = %d has Name %s\n", gorilla.ID, gorilla.Name)
+	} else {
+		fmt.Println(err)
+	}
+
+	newAnimals := Sector{
+		Subtype: "animals",
+		Animals: []Animal{
+			{ID: 10, Name: "Cow"},
+		},
+	}
+
+	ungulatesAnimals := z.Areas["ungulates"].Sectors["animals"]
+	ungulatesAnimals.MoveAnimal(2, &newAnimals)
+
+	z.Areas["ungulates"].Sectors["newAnimals"] = newAnimals
+
+	fmt.Println("Animals from new sector")
+	for _, animal := range z.Areas["ungulates"].Sectors["newAnimals"].Animals {
+		fmt.Printf("%s found, animal ID %d\n", animal.Name, animal.ID)
+	}
+}
+
+func FindAnimalByName(areas Areas, name string) (*Animal, error) {
```

Comment:
Чи можемо ми порефакторити код, щоб працювало таке?
```suggestion
func (z *Zoo) FindAnimalByName(name string) (*Animal, error) {
```

Reply:
і тут